### PR TITLE
Handle season generation in python instead of sql

### LINKED ISSUE #58: Decouple code and data for season generation
The current way of generating seasons couples the code to the data inserted in the database and if they get out of sync we run into issues. It should just copy the setup of the previous season instead of having so much hardcoded stuff. Then you can change the setup of future seasons by just inserting data manually, and the code knows how to handle it from there.

### PR DESCRIPTION
**Pull request by BlackYps**
Closes #58 

We can now set the season duration from the config

### CODE REVIEW

File: service/season_generator.py [hunk 1]
```diff
@@ -40,6 +43,121 @@ async def check_season_end(self):
...
+            year = now.year
+            month = now.month
+            if month == 12:
+                month = 1
+                year += 1
+            else:
+                month += 1
+            # season starts and ends at noon, so that all timezones see the same date in the client
+            start_date = datetime(year=year, month=month, day=1, hour=12)
+            end_date = start_date + relativedelta(months=SEASON_LENGTH_MONTHS) - relativedelta(days=1)
```
Comment (Askaholic) [inline service/season_generator.py:53]: Is `relativedelta` doing anything different from the builtin `timedelta`?

Comment (Askaholic) [reply service/season_generator.py:53]: Ah I see. It has months.

File: service/season_generator.py [hunk 2]
```diff
@@ -40,6 +43,121 @@ async def check_season_end(self):
...
+            rows = await result.fetchall()
+
+            now = datetime.now()
+            year = now.year
+            month = now.month
+            if month == 12:
+                month = 1
+                year += 1
+            else:
+                month += 1
```
Comment (Askaholic) [inline service/season_generator.py]: This code is just trying to add one month to the date right? How about just using a `relativedelta(months=1)`?
```suggestion
            now = datetime.now() + relativedelta(months=1)
            year = now.year
            month = now.month
```

Then you probably don't even need to assign `year` and `month` to variables

Comment (BlackYps) [reply service/season_generator.py]: Good idea. I think I did this before I discovered the relativedelta package.

File: service/season_generator.py [hunk 3]
```diff
@@ -40,6 +43,114 @@ async def check_season_end(self):
...
+        )
+        await conn.execute(season_insert_sql)
+
+        division_sql = (
+            select([league_season_division])
+            .where(league_season_division.c.league_season_id == season_row[league_season.c.id])
+        )
+        result = await conn.execute(division_sql)
+        season_division_rows = await result.fetchall()
+        if len(season_division_rows) == 0:
```
Comment (Askaholic) [inline service/season_generator.py]: The pythonic way to check for an empty list is like this.
```suggestion
        if not season_division_rows:
```

File: service/season_generator.py [hunk 4]
```diff
@@ -40,6 +43,114 @@ async def check_season_end(self):
...
+            await conn.execute(division_insert_sql)
+
+            subdivision_sql = (
+                select([league_season_division_subdivision])
+                .where(league_season_division_subdivision.c.league_season_division_id ==
+                       division_row[league_season_division.c.id])
+            )
+            result = await conn.execute(subdivision_sql)
+            subdivision_rows = await result.fetchall()
+            if len(subdivision_rows) == 0:
```
Comment (Askaholic) [inline service/season_generator.py]: Same

**Review by Askaholic — APPROVED** [review]

**Comment by BlackYps** [comment]
Thank you for your review!

**Pull request merged**